app/org/sagebionetworks/bridge/play/controllers/BaseController.java

```diff
@@ -258,13 +260,34 @@ void verifySupportedVersionOrThrowException(Study study) throws UnsupportedVersi
                 LOG.debug("Malformed Accept-Language header sent: " + acceptLanguageHeader);
             }
         }
+
+        // if no Accept-Language header detected, we shall add an extra warning header
+        Http.Response response = Http.Context.current().response();
+        if (response.getHeaders().containsKey(BridgeConstants.BRIDGE_API_STATUS_HEADER)) {
+            String previousWarning = response.getHeaders().get(BridgeConstants.BRIDGE_API_STATUS_HEADER);
+            response.setHeader(BridgeConstants.BRIDGE_API_STATUS_HEADER, previousWarning + "; " + BridgeConstants.WARN_NO_ACCEPT_LANGUAGE);
+        } else {
+            response.setHeader(BridgeConstants.BRIDGE_API_STATUS_HEADER, BridgeConstants.WARN_NO_ACCEPT_LANGUAGE);
+        }
         return new LinkedHashSet<>();
     }
     
     ClientInfo getClientInfoFromUserAgentHeader() {
         String userAgentHeader = request().getHeader(USER_AGENT);
         ClientInfo info = ClientInfo.fromUserAgentCache(userAgentHeader);
-        
+
+        // if the user agent cannot be parsed (probably due to missing user agent string or unrecognizable user agent),
+        // should set an extra header to http response as warning - we should have an user agent info for filtering to work
+        if (info.equals(ClientInfo.UNKNOWN_CLIENT)) {
+            Http.Response response = Http.Context.current().response();
+            if (response.getHeaders().containsKey(BridgeConstants.BRIDGE_API_STATUS_HEADER)) {
+                String previousWarning = response.getHeaders().get(BridgeConstants.BRIDGE_API_STATUS_HEADER);
+                response.setHeader(BridgeConstants.BRIDGE_API_STATUS_HEADER, previousWarning + "; " + BridgeConstants.WARN_NO_USER_AGENT);
+            } else {
+                response.setHeader(BridgeConstants.BRIDGE_API_STATUS_HEADER, BridgeConstants.WARN_NO_USER_AGENT);
+            }
+        }
```

Comment:
Can you make a method, addWarningToHeader(String warning), to consolidate this logic?